src/bp-groups/bp-groups-functions.php

```diff
@@ -608,6 +608,21 @@ function groups_leave_group( $group_id, $user_id = 0 ) {
 		}
 	}
 
+	// If group restrict invites is enabled and any member left parent group then also remove from all child groups.
+	if (
+		! empty( $group_id ) &&
+		true === bp_enable_group_hierarchies() &&
+		true === bp_enable_group_restrict_invites() ) {
+		$groups = bp_get_descendent_groups( $group_id, $user_id );
+		if ( ! empty( $groups ) ) {
+			foreach ( $groups as $group ) {
+				if ( $group->is_member ) {
+					groups_leave_group( $group->id, $user_id );
+				}
+			}
+		}
+	}
```

Comment:
move this code below to remove the member here: https://github.com/buddyboss/buddyboss-platform/pull/4153/files#diff-49a7692f12ab8cd369c8accbad36f61734b28e5b080458bb1decfd1b11b36abfR629